src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java

```diff
@@ -1040,7 +1006,33 @@ private void updateQueryableProperties(
                 valueLoader = new ConfigLoader(id);
             }
 
-            info = cache.get(id, valueLoader);
+            Semaphore lock = locks.get(id);
+            if (lock == null) {
+                lock = new Semaphore(1);
+                locks.put(id, lock);
+            }
```

Comment:
It probably wouldn't matter much but this code block (and below in acquireWriteLock) can be performed atomically using: java.util.concurrent.ConcurrentHashMap.computeIfAbsent(K, Function<? super K, ? extends V>)

Semaphore lock = locks.computeIfAbsent(id, x -> new Semaphore(1));

Reply:
That is very elegant, thanks for the tip because I use that pattern a lot and wasn't aware yet of this :)

Reply:
I changed it.

Reply:
You changed it down in acquireWriteLock but not up here.

Reply:
right, you are sharp :) fixed